code/modules/jobs/job_types/_job.dm

```diff
@@ -173,8 +186,27 @@
 	return TRUE
 
 /datum/job/proc/map_check()
+	var/list/job_changes = GetMapChanges()
+	if(!job_changes)
+		return FALSE
 	return TRUE
 
+/**
+  * Gets the changes dictionary made to the job template by the map config. Returns null if job is removed.
+  */
+/datum/job/proc/GetMapChanges()
+	var/string_type = "[type]"
+	var/list/splits = splittext(string_type, "/")
```

Comment:
Why by type instead of name ?

Reply:
By going by type we can also choose to rename the job in the map as an extension to this system. I just used the end part to reduce redundancy in the JSON.

Reply:
using the end part only might cause issues further ahead
things like `/datum/job/detective/allied_faction` and `/datum/job/assistant/allied_faction` will end up the same
is this entirely necessary?